certs.go

```diff
@@ -225,6 +221,21 @@ func writeFile(file string, contents []byte) error {
 	return err
 }
 
+func getDNSNames(setNames []string) []string {
+	dnsNames := []string{"localhost"}
+	hostname, err := os.Hostname()
+	if err == nil {
+		// If there's a hostname for the local machine, add it to the cert's DNS names.
+		dnsNames = append(dnsNames, hostname)
+	}
+
+	// if DNSNames specified use only the specified DNS Names list
+	if len(setNames) > 0 {
```

Comment:
This block should go first. If the user provided DNS names we don't need to look at `os.Hostname()` or do any other work.